lib/model/message_list.dart

```diff
@@ -682,6 +848,18 @@ class MessageListView with ChangeNotifier, _MessageSequence {
           notifyListeners();
           fetchInitial();
         }
+
+        outboxMessages.clear();
+        for (final outboxMessage in store.outboxMessages.values) {
+          if (_shouldAddOutboxMessage(
+            outboxMessage,
+            wasUnmuted: outboxMessage is StreamOutboxMessage
+              && outboxMessage.conversation.streamId == event.streamId
```

Comment:
I think this whole stanza goes away if we let `fetchInitial` take care of syncing in the outbox messages when appropriate.